tests/tt_eager/python_api_testing/unit_testing/test_moreh_norm.py

```diff
@@ -0,0 +1,227 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+
+import tt_lib as ttl
+from models.utility_functions import comp_allclose, skip_for_wormhole_b0
+from loguru import logger
+
+TILE_HEIGHT = 32
+TILE_WIDTH = 32
+
+
+def to_cpu(npu_tensor, shape, *, cpu_layout=ttl.tensor.Layout.ROW_MAJOR):
+    if npu_tensor is None:
+        return None
+    cpu_tensor = npu_tensor.cpu().to(cpu_layout).unpad_from_tile(shape).to_torch()
+    return cpu_tensor
+
+
+def to_npu(
+    cpu_tensor,
+    device,
+    *,
+    npu_layout=ttl.tensor.Layout.TILE,
+    npu_dtype=ttl.tensor.DataType.BFLOAT16,
+):
+    if cpu_tensor is None:
+        return None
+    npu_tensor = ttl.tensor.Tensor(cpu_tensor, npu_dtype).pad_to_tile(float("nan")).to(npu_layout).to(device)
+    return npu_tensor
+
+
+def make_cpu_tensors(input_shape, dim):
+    if dim is None:
+        dim = list(range(len(input_shape)))
+
+    if isinstance(dim, int):
+        dim = [dim]
+
+    # output_grad_shape
+    output_grad_shape = input_shape[:]
+
+    for d in dim:
+        output_grad_shape[d] = 1
+
+    # input
+    cpu_input = torch.empty(input_shape, dtype=torch.float32).uniform_(-1, 1).requires_grad_()
+
+    # output_grad
+    cpu_output_grad = torch.empty(output_grad_shape, dtype=torch.float32).uniform_(-1, 1)
+
+    return cpu_input, cpu_output_grad
+
+
+def torch_norm(cpu_x, cpu_dy, *, p=2.0, dim=None, do_backward=False):
+    cpu_y = torch.norm(cpu_x, p=p, dim=dim, keepdim=True)
+
+    cpu_dx = None
+    if do_backward:
+        cpu_y.backward(cpu_dy)
+        cpu_dx = cpu_x.grad
+
+    return cpu_y, cpu_dx
+
+
+def tt_norm(cpu_x, cpu_dy, *, p=2.0, dim=None, do_backward=False, device=None):
+    npu_x = to_npu(cpu_x.bfloat16(), device)
+    if do_backward:
+        npu_dy = to_npu(cpu_dy.bfloat16(), device)
+
+    npu_y = ttl.operations.primary.moreh_norm(npu_x, p=p, dim=dim)
```

Comment:
> I think it's better to use `cpu_y`. This is because the floating point error is larger when using npu_y.

This part is needed for the forward test.